packages/babel-helpers/src/helpers.js

```diff
@@ -399,23 +399,31 @@ helpers.inherits = () => template.program.ast`
     if (typeof superClass !== "function" && superClass !== null) {
       throw new TypeError("Super expression must either be null or a function");
     }
-    setPrototypeOf(subClass.prototype, superClass && superClass.prototype);
+    subClass.prototype = Object.create(superClass && superClass.prototype, {
+      constructor: {
+        value: subClass,
+        enumerable: false,
+        writable: true,
+        configurable: true
+      }
+    });
     if (superClass) setPrototypeOf(subClass, superClass);
   }
 `;
 
 helpers.inheritsLoose = () => template.program.ast`
   export default function _inheritsLoose(subClass, superClass) {
-    subClass.prototype.__proto__ = superClass && superClass.prototype;
+    subClass.prototype = Object.create(superClass.prototype);
+    subClass.prototype.constructor = subClass;
     subClass.__proto__ = superClass;
   }
 `;
 
 helpers.getPrototypeOf = () => template.program.ast`
   export default function _getPrototypeOf(o) {
-    _getPrototypeOf = Object.getPrototypeOf || function _getPrototypeOf(o) {
-      return o.__proto__;
-    };
+    _getPrototypeOf = Object.setPrototypeOf
+      ? Object.getPrototypeOf
+      : function _getPrototypeOf(o) { return o.__proto__; };
```

Comment:
I think I'd still want to fall back to `Object.getPrototypeOf` if `__proto__` falsy, since that was the old behavior.

Reply:
Done.